File: FacebookShortcut/app/src/main/java/com/facebook/facebookvr/MainActivity.java
package com.facebook.facebookvr;

import android.app.Activity;
import android.content.Intent;
import android.net.Uri;

public class MainActivity extends Activity {
    private boolean openedWeb = false;
    private boolean launch() {
        Intent launchIntent = getPackageManager().getLaunchIntentForPackage("com.threethan.launcher");
        if (launchIntent != null) return launchIt(launchIntent);
        return false;
    }
    private boolean launchIt(Intent launchIntent) {
        launchIntent.setFlags(Intent.FLAG_ACTIVITY_NO_ANIMATION
                | Intent.FLAG_ACTIVITY_REORDER_TO_FRONT);
        finish();
        startActivity(launchIntent);
        return true;
    }

    @Override
    public void onResume() {
        super.onResume();
        if (!launch() && !openedWeb) {
            openedWeb = true;
            final String downloadUrl = "https://github.com/threethan/LightningLauncher/releases/";
            Intent browserIntent = new Intent(
                    Intent.ACTION_VIEW,
                    Uri.parse(downloadUrl)
            );
            startActivity(browserIntent);
        }
    }
}

File: LauncherShortcut/app/src/main/java/com/facebook/orca/MainActivity.kt
package com.facebook.orca

import android.app.Activity
import android.content.Intent
import android.net.Uri

class MainActivity : Activity() {
    private val downloadUrl = "https://github.com/threethan/LightningLauncher/releases/"
    private var openedWeb = false
    private fun launch(): Boolean {
        var launchIntent = packageManager.getLaunchIntentForPackage("com.threethan.launcher")
        launchIntent?.let { return launchIt(it); }
        launchIntent = packageManager.getLaunchIntentForPackage("com.basti564.dreamgrid")
        launchIntent?.let { return launchIt(it); }
        launchIntent = packageManager.getLaunchIntentForPackage("com.veticia.piLauncherNext")
        launchIntent?.let { return launchIt(it); }
        return false
    }
    private fun launchIt(launchIntent: Intent): Boolean {
        launchIntent.flags = Intent.FLAG_ACTIVITY_NO_ANIMATION
        finish()
        startActivity(launchIntent)
        return true
    }

    override fun onResume() {
        super.onResume()
        if (!launch() && !openedWeb) {
            openedWeb = true
            val browserIntent = Intent(
                Intent.ACTION_VIEW,
                Uri.parse(downloadUrl)
            )
            startActivity(browserIntent)
        }
    }
}